lib/common/api/deprecate.js

```diff
@@ -67,6 +67,25 @@ const deprecate = {
     })
   },
 
+  promisify: (promise, cb) => {
```

Comment:
This as an API would be used something like this

```js
const originalMethod = app.myMethod;
app.myMethod = (a, b, c, cb) => {
  return deprecate.promisify(originalMethod(a, b, c), cb);
}
```

It might be a nicer API if we have this

```js
// deprecate.callbackToPromise(module, methodName, callbackIndex);
deprecate.callbackToPromise(app, 'myMethod', 3);
```